src/accelerate/accelerator.py

```diff
@@ -1989,26 +2009,31 @@ def _prepare_ipex_or_xpu(self, *args):
                 result[i] = optimizer
         return tuple(result)
 
-    def _prepare_msamp(self, *args):
+    def _prepare_msamp(self, *args, device_placement):
         if not is_msamp_available():
             raise ImportError(
                 "MS-AMP was not found on your system. Please ensure that MS-AMP is available "
                 " or choose `'te'` as the backend for FP8 mixed precision training."
             )
-        else:
+            # We've already checked for FSDP + MS-AMP during `__init__`
```

Comment:
Wait, by removing the `else`, `msp` is only imported `if not is_msamp_available()`?